openedx/adg/admin_dashboard/views.py

```diff
@@ -0,0 +1,55 @@
+"""
+Contains all the views related code that'll be used for the admin_dashboard package and any apps it contains
+"""
+from django.contrib.auth.mixins import AccessMixin
+from django.contrib.auth.views import redirect_to_login
+from django.http import HttpResponse
+from django.views.generic import TemplateView
+from rest_framework.status import HTTP_400_BAD_REQUEST
+
+from openedx.adg.lms.applications.helpers import has_admin_permissions
+
+
+class LoginRequiredAndUserTestPassesMixin(AccessMixin):
```

Comment:
If the name of this template is inspired by the mixin provided by the `django-braces` package, then the name should be `LoginRequiredAndUserPassesTestMixin` because the name of the package provided by django-braces is [`UserPassesTestMixin`](https://django-braces.readthedocs.io/en/latest/access.html#userpassestestmixin) not `UserTestPassesMixin`

Reply:
UserPassesTestMixin is provided by Django not `django-braces`. Also, it was not inspired by that directly, but I feel like that is a good idea and I'll change the name accordingly.

Reply:
It's apparently also provided by braces, if you open the linked resource in my comment:

`from braces.views import UserPassesTestMixin`

Reply:
Yes braces basically created a bunch of mixins on top of that. It is a userful package.
I was just saying that Django provides it out of the box